src/pfe/build

```diff
@@ -1,6 +1,7 @@
-mvn -f iterative-dev/pom.xml clean package -DskipTests | fgrep -v 'Downloaded' | fgrep -v 'Downloading';
-if [ $? -ne 0 ]; then
-    echo "iterative-dev build failed"
+mvn -f iterative-dev/pom.xml clean package -DskipTests
```

Comment:
Wondering don't we need this `| fgrep -v 'Downloaded' | fgrep -v 'Downloading';` part anymore?

I was thinking that adding `set -e` on top of the script which will exit, if any of the commands fail

Reply:
`set -o pipefail` specifically maybe helpful here

Reply:
`set -e` didnt work. Idk why we need to search for Downloaded or Downloading in mvn builds, seems pointless to me?

If mvn is successful it will give EC 0, otherwise it will give EC 1.

Reply:
Yeah `set -e` wasn't the correct command. `set -o pipefail` should work though

Reply:
Make sure to put it on top of the script

Reply:
pipefail works on the terminal run.sh ie it can detect errors but it never works on Jenkins build

always gives me
```
14:54:04  +++   BUILDING DOCKER IMAGES   +++
14:54:04  
14:54:04  Building image codewind-pfe-amd64
14:54:04  build: 1: set: Illegal option -o pipefail
14:54:04  
14:54:04  real	0m0.001s
14:54:04  user	0m0.001s
14:54:04  sys	0m0.000s
14:54:04  +++   FAILED TO BUILD codewind-pfe-amd64 - exiting.   +++

```

Reply:
The other option to use is `${PIPESTATUS[0]}` but i dont see the point in this when i am always getting the exit code of the first cmd in pipe, what is the point in piping then?

Reply:
Tbh, I am fine with `fgrep -v 'Downloaded' | fgrep -v 'Downloading';` being removed as well if we don't need it. I don't know why those were in the script to begin with

Reply:
`fgrep -v 'Downloaded' | fgrep -v 'Downloading';` was added in Microclimate to remove the maven "Download", "Downloading" messages that were getting spammed to the build logs whenever Microclimate was built.

Not sure if that's needed any more though, I pulled down Maysun's branch and still don't see `Download` or `Downloading` when I build.